src/server/system_services/master_key_manager.js

```diff
@@ -132,10 +169,16 @@ class MasterKeysManager {
      * @returns {Object}  
      */
     _resolve_master_key(m_key) {
+        const { NOOBAA_ROOT_SECRET } = process.env;
+        // in case we are resolving an old structured root-key, we will update to the new format
+        if (!NOOBAA_ROOT_SECRET && this.is_root_key(m_key.master_key_id) && !m_key.root_key_id) {
+            m_key.root_key_id = this.get_current_root_key_id();
+            m_key.master_key_id = undefined;
```

Comment:
why setting m_key.master_key_id = undefined; don't use it even after the update to the new form?